### DISCUSSION TITLE
Erc20 transfer transactions

### DISCUSSION THREAD

**Discussion by LidiaMil** [L-PAY/EVM_Parser#5] [Q&A] [answered]
Why don't we parse erc20 transfer event directly from transaction struct?

**Comment by masxxiii** [comment, marked as answer]
We can directly parse the data we need from the transfer event of a transaction. However, not all erc20 transfer transactions have only one event. If you use a centralised wallet like Huobi to transfer erc20 tokens to yourself, you will see that the transaction might have multiple transfer events and only one of those events are related to erc20 tokens being transferred to your wallet. This is because centralised wallets use special smart contracts to transfer erc20 tokens to different wallets to save gas instead of directly calling the Transfer method on an erc20 token contract.

**Reply by stephmcl18** [reply]
Hello can you explain to me how to get my erc20 tokens off of the etherium blockchain and onto a wallet so i can then access the amount of money that my token holds? Ive been trying to figure this out for a week now and i can not figure it out...

**Reply by masxxiii** [reply]
You can use a third party API service like Etherscan. They have dedicate API for your use case. Here is the link to their API Documentation. https://docs.etherscan.io/api-endpoints/tokens

However, if you would like to build a custom solution, there are certain ways you can approach this problem. One simple approach is to have a list of ERC20 tokens and batch request the balance of your wallet from the contracts and then filter out the tokens that have balance greater than zero. Hope this help!
